### PR TITLE
[INTEL MKL] MKL ML code cleanup -- mkl_util

### PR DESCRIPTION
**Pull request by gzmkl**
Clean up MKL ML code from mkl_util.h and mkl_util_test.cc.   (Remove ~800 lines of unused code)

MKL ML implementation is no longer supported.

### CODE REVIEW

File: tensorflow/core/util/mkl_util.h [hunk 1]
```diff
@@ -83,7 +59,6 @@ namespace tensorflow {
 // MKL operation, and did not go through a conversion to a standard
 // Tensorflow tensor.
 
-// For use with MKL ML, has been deprecated
 typedef enum { W = 0, H = 1, C = 2, N = 3 } MklDims;
```
Comment (penpornk) [inline tensorflow/core/util/mkl_util.h]: If this is only used in MKL-ML (according to the comment above), should it be removed as well?

File: tensorflow/core/util/mkl_util.h [hunk 2]
```diff
@@ -1974,9 +1354,9 @@ class MklDnnData {
     return false;
   }
 
-  /// TODO: this is a faster path with reorder primitive cache compared with
-  /// CheckReorderToOpMem(..., std::vector<primitive>* net), will remove
-  /// slow path in the future
+  /// This is a faster path with reorder primitive cache compared with
+  /// CheckReorderToOpMem(..., std::vector<primitive>* net).
+  /// The slower path will be removed in the future
```
Comment (penpornk) [inline tensorflow/core/util/mkl_util.h]: Adding TODO for easy search.
```suggestion
  /// TODO: Remove the slower path.
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the cleanup!

File: tensorflow/core/util/mkl_util.h (hunk 1 above)
Comment (gzmkl) [reply tensorflow/core/util/mkl_util.h]: Good point, let me double-check. Thanks!

File: tensorflow/core/util/mkl_util.h (hunk 2 above)
Comment (gzmkl) [reply tensorflow/core/util/mkl_util.h]: Will take the suggestion

File: tensorflow/core/util/mkl_util.h (hunk 1 above)
Comment (gzmkl) [reply tensorflow/core/util/mkl_util.h]: Done by removing the enum and related deprecated method GetMklTensorDimIndex.

File: tensorflow/core/util/mkl_util.h (hunk 2 above)
Comment (gzmkl) [reply tensorflow/core/util/mkl_util.h]: Done

**Review by penpornk — APPROVED** [review]
Thank you very much!

**Pull request merged**